contrib/mesos/pkg/executor/service/service.go

```diff
@@ -200,7 +225,19 @@ func (s *KubeletExecutorServer) runKubelet(
 	}()
 
 	// create main pod source, it will stop generating events once executorDone is closed
-	newSourceMesos(executorDone, kcfg.PodConfig.Channel(mesosSource), podLW, registry)
+	var containerOptions []podsource.Option
+	if s.containerID != "" {
+		// tag all pod containers with the containerID so that they can be properly GC'd by Mesos
+		containerOptions = append(containerOptions, podsource.ContainerEnvOverlay([]api.EnvVar{
```

Comment:
I do like the simplicity of this :) makes me question my motives for `PrependFilter` (which requires this class to understand podsource filters at a deeper level)